grammar/haxe.bnf

```diff
@@ -254,11 +254,12 @@ private topLevelDeclaration ::= classDeclaration
 
 privateKeyWord ::= 'private'
 externOrPrivate ::= 'extern' | privateKeyWord
+private externAndMaybePrivate ::= privateKeyWord? 'extern' privateKeyWord?
 
 typedefDeclaration ::= macroClassList? externOrPrivate? 'typedef' componentName genericParam? '=' functionTypeWrapper ';'?
 {pin=5 mixin="com.intellij.plugins.haxe.lang.psi.impl.AbstractHaxeTypeDefImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeClass"}
 
-externClassDeclaration ::= macroClassList? externOrPrivate* ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
+externClassDeclaration ::= macroClassList? externAndMaybePrivate ('class' | 'interface') componentName genericParam? inheritList? '{' externClassDeclarationBody '}'
```

Comment:
Do you mean allow extern modifier for `interfaceDeclaration`(as it used before this pull request), and remove `interface` from `externalClassDeclaration`? (additionally add check for AbstractHaxePsiClass.java/isExtern function for external interface?)
I not sure about external interface, but I think it should be close to regular interface, just with public fields
I read about those pins, and if I got it right, then I agree with you, we can use pin=3.